zrml/court/src/lib.rs

```diff
@@ -188,315 +170,1166 @@ mod pallet {
+            Self::check_appealable_market(&market_id, &court, now)?;
+
+            // the outcome which would be resolved on is appealed (including oracle report)
+            let old_draws = Draws::<T>::get(market_id);
+            let appealed_outcome =
+                Self::get_latest_resolved_outcome(&market_id, old_draws.as_slice())?;
+            let appeal_info = AppealInfo { backer: who.clone(), bond, appealed_outcome };
+            court.appeals.try_push(appeal_info).map_err(|_| {
+                debug_assert!(false, "Appeal bound is checked above.");
+                Error::<T>::MaxAppealsReached
+            })?;
+
+            let new_draws = Self::select_jurors(&market_id, old_draws, appeal_number)?;
+
+            let last_resolve_at = court.periods.appeal_end;
+            let _ids_len_0 = T::DisputeResolution::remove_auto_resolve(&market_id, last_resolve_at);
+
+            let request_block = <RequestBlock<T>>::get();
+            debug_assert!(request_block >= now, "Request block must be greater than now.");
+            let pre_vote_end = request_block.saturating_sub(now);
+            let periods = Periods {
+                pre_vote_end,
+                vote_end: T::CourtVotePeriod::get(),
+                aggregation_end: T::CourtAggregationPeriod::get(),
+                appeal_end: T::CourtAppealPeriod::get(),
+            };
+            // sets periods one after the other from now
+            court.update_periods(periods, now);
+            let new_resolve_at = court.periods.appeal_end;
+            debug_assert!(new_resolve_at != last_resolve_at);
+            let _ids_len_1 =
+                T::DisputeResolution::add_auto_resolve(&market_id, court.periods.appeal_end)?;
+
+            T::Currency::reserve_named(&Self::reserve_id(), &who, bond)?;
+
+            <Courts<T>>::insert(market_id, court);
+            <Draws<T>>::insert(market_id, new_draws);
+
+            let appeal_number = appeal_number as u32;
+            Self::deposit_event(Event::MarketAppealed { market_id, appeal_number });
+
+            Ok(())
         }
 
-        // No-one can stake more than BalanceOf::<T>::max(), therefore, this function saturates
-        // arithmetic operations.
-        fn current_required_stake(jurors_num: usize) -> BalanceOf<T> {
-            let jurors_len: BalanceOf<T> = jurors_num.saturated_into();
-            T::StakeWeight::get().saturating_mul(jurors_len)
+        /// Back the global dispute to allow the market to be resolved
+        /// if the last appeal outcome is disagreed on.
+        ///
+        /// # Arguments
+        ///
+        /// - `market_id`: The identifier of the court.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(1)`
+        #[pallet::weight(1_000_000_000_000)]
+        #[transactional]
+        pub fn back_global_dispute(
+            origin: OriginFor<T>,
+            market_id: MarketIdOf<T>,
+        ) -> DispatchResult {
```

Comment:
The extrinsic is a way to have a financial cost to appeal the last resolvable outcome. But this last appeal is different to the previous appeals, because we don't have the functionality to start a new court round (selecting jurors and so on..).

With adding the last appeal, the `appeals` BoundedVec is full and we can check that inside `has_failed` with `appeals.is_full()`. If the appeals is full, the global dispute is able to get triggered.

Reply:
I removed the extrinsic `back_global_dispute` and merged it into `appeal`.